src/helpers/ProtocolMetricsHelper.ts

```diff
@@ -0,0 +1,404 @@
+import get from "get-value";
+import { CSSProperties } from "react";
+import { ChartType } from "src/components/Chart/Constants";
+
+import { getFloat } from "./NumberHelper";
+
+export type TokenRow = {
+  token: string;
+  category: string;
+  value: string;
+};
+
+type TokenMap = {
+  [key: string]: TokenRow;
+};
+
+export const objectHasProperty = (object: unknown, property: string): object is Record<string, unknown> => {
+  return typeof object === "object" && object !== null && property in object;
+};
+
+/**
+ * As the subgraph data source takes a long time to re-index, we alter
+ * the display name of certain tokens here.
+ *
+ * This should be used sparingly, and the updated display names should be integrated
+ * into the subgraph ASAP.
+ */
+export const renameToken = (value: string): string => {
+  const tokenMap = new Map<string, string>([
+    ["Uniswap V2 OHM V2-DAI Liquidity Pool", "SushiSwap OHM-DAI Liquidity Pool"],
+    ["Uniswap V2 OHM V2-ETH Liquidity Pool", "SushiSwap OHM-ETH Liquidity Pool"],
+    ["cvxOHMETH", "Curve OHM-ETH Pool (Staked in Convex)"],
+    ["Balancer OHM-DAI-WETH Pool", "Balancer V2 OHM-DAI-WETH Liquidity Pool"],
+    ["Balancer WETH-FDT Pool", "Balancer V2 WETH-FDT Liquidity Pool"],
+    ["vlCVX V2", "vlCVX"],
+  ]);
+
+  const mapValue = tokenMap.get(value);
+  if (!mapValue) return value;
+
+  return mapValue;
+};
+
+type ComponentRecordRow = {
+  token: string;
+  value: string;
+};
+
+type Components = {
+  value: number;
+  records: ComponentRecordRow[];
+  tokens: TokenMap;
+};
+
+export type MetricComponentsTokenSummary = {
+  treasuryLiquidBackingProtocolOwnedLiquidityComponents?: Components;
+  treasuryLiquidBackingStableComponents?: Components;
+  treasuryLiquidBackingVolatileComponents?: Components;
+  treasuryLPValueComponents?: Components;
+  treasuryStableValueComponents?: Components;
+  treasuryVolatileValueComponents?: Components;
+};
+
+export type BaseMetric = {
+  timestamp: number;
+} & Record<string, unknown>;
+
+/**
+ * For each of the {keys} specified, calculates a total for each token.
+ *
+ * For example:
```

Comment:
```suggestion
 * For example:
 * @example
```

Reply:
just a dumb nit, not a big deal. If you think `For example` looks cleaner then we can leave it.

Reply:
Added

Reply:
Done